fitbenchmarking/fitbenchmarking
# Issue #477: Refactor logger

#### Description of Work

Fixes #477 

Refactors the logger to be more explicit in usage.
Changes all print calls to logger calls.
Some calls may need re-classifying.


#### Testing Instructions

1. Run with different logging options, check if information is as you'd expect.

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
